NASA-AMMOS/aerie-ui
# Implement Permissions on Scheduling Page

Resolves #730 
Depends on https://github.com/NASA-AMMOS/aerie/pull/1031

This adds permissions checks for creating/editing/deleting scheduling goals and conditions on the scheduling page. I was originally also going to include the work for the scheduling panes, but decided to split that into a separate PR to make this easier to review.

Testing this will follow a similar pattern as previous permissions tasks. What we want to confirm is that users can only create goals or conditions in plans they own or collaborate on. And goals/conditions can only be edited by those who are owners/collaborators in the associated plan.

To test:

1. Upload two models
2. Create a few plans using each model
3. Go to http://localhost:8080/console/data/AerieMerlin/schema/public/tables/plan/browse and change one plan's owner to be the owner that you want to test with
4. Log in as that owner and switch to the "user" role from the top right dropdown
5. Navigate to http://localhost:3000/scheduling
6. Create a new scheduling goal
7. Verify that only plans that your user is owner of are selectable from the "Plan" dropdown
8. Verify that only models that have instances of plans that your user is owner of are selectable from the "Model" dropdown
9. Navigate back to http://localhost:3000/scheduling
10. Verify that the goal that you created is editable
11. Log out and log in with a different username and switch to the "user" role again
12. Navigate back to http://localhost:3000/scheduling again
13. Verify that the goal that you previously created is no longer editable/deletable
14. Navigate to the edit view of that constraint by manually going to: http://localhost:3000/scheduling/goals/edit/[CONSTRAINT ID]
15. Verify that everything is read only

Repeat above steps for scheduling conditions.
